### ISSUE TITLE
guide

### ISSUE THREAD

**Issue by tony-xlh** [tony-xlh/APEX-Dynamic-Web-TWAIN#3]
Simple Guide using the Web TWAIN plugin:

1. Download the plugin and install it: https://github.com/tony-xlh/APEX-Dynamic-Web-TWAIN/raw/main/region_type_plugin_com_dynamsoft_dwt.sql
    
    ![image](https://github.com/tony-xlh/APEX-Dynamic-Web-TWAIN/assets/112376616/b664b9f5-4c64-478f-99e9-4fa5a333e1fa)

2. Create a new page and drag the document scanner region into it. 

    ![image](https://github.com/tony-xlh/APEX-Dynamic-Web-TWAIN/assets/112376616/89fde17e-2d78-4dd6-8143-30145daf1e59)

3. Set the region’s ID to DWT. 

    ![image](https://github.com/tony-xlh/APEX-Dynamic-Web-TWAIN/assets/5462205/3c1fe9f4-8bab-4a3a-870d-7a5a8899402a)

4. Create a new button with a dynamic action to start the scanner.  It executes the following code: apex.region("DWT").showModal();

    ![image](https://github.com/tony-xlh/APEX-Dynamic-Web-TWAIN/assets/112376616/3f7a8086-06df-427c-9bf2-e2a7ae50d923)

5. Run the app and you will be able to scan documents.

    ![image](https://github.com/tony-xlh/APEX-Dynamic-Web-TWAIN/assets/112376616/923320bc-ba59-44ea-aa13-39418e9d47fe)

6. You can set several attributes for the component.

    ![image](https://github.com/tony-xlh/APEX-Dynamic-Web-TWAIN/assets/112376616/523b4196-8a05-47a8-85da-2d19392c11d8)

    * You have to set the license attribute to use Dynamic Web TWAIN in a production environment. You can apply for a license [here](https://www.dynamsoft.com/customer/license/trialLicense?product=dwt).
    * If you need to upload the documents, you have to start an HTTP server to receive the documents and set the host and port attributes.

**Comment by xulihang** [comment]
## Notes about Version

The latest plugin was exported using APEX 24 which may not work with old APEX versions.

You can use this old version instead: https://github.com/tony-xlh/APEX-Dynamic-Web-TWAIN/raw/c93509b21cf51de0ef60ffef34d5d247c5e65e8d/region_type_plugin_com_dynamsoft_dwt.sql

Or create a new plugin from scratch following the blog.

**Comment by AbdulMueed209** [comment]
Hi Team,
Can we implement this plugin in cloud Oracle Apex Workspace
[Oracle Apex Workspace Link](https://apex.oracle.com/pls/apex/f?p=4550:1::::::)
If yes then please guide what values we will set in Plugin's attribute

**Comment by xulihang** [comment]
Yes. The online demo runs on cloud Oracle APEX: https://apex.oracle.com/pls/apex/r/dynamsoft/dynamsoft-demos/document-scanner

**Comment by AbdulMueed209** [comment]
This error raised while posting image to server
![image](https://github.com/user-attachments/assets/223312cb-a2f7-436d-9e49-c5eff8870c9a)


Please guide about attributes of plugin region when I use this plugin to Cloud Workspace.
Currently I use that attribute for plugin

![image](https://github.com/user-attachments/assets/0dc30c8a-7965-47ee-bf44-216469a03ddf)

**Comment by xulihang** [comment]
It's explained in the guide above.

**Comment by xulihang** [comment]
You can also check out the blog to learn more: https://www.dynamsoft.com/codepool/oracle-apex-document-scanning.html

**Comment by AbdulMueed209** [comment]
OK DONE.
Thanks.

**Comment by AbdulMueed209** [comment]
Hi Experts,

I am facing an issue with creating and using a plugin. The plugin relies on an online link to the dwt.azurewebsites.net over port 443. However, I need to deploy this plugin on a server that does not have internet access.

Is there a way to host this plugin locally on the existing server?

![WhatsApp Image 2024-12-22 at 16 27 11_16b81410](https://github.com/user-attachments/assets/7d6a1759-a992-46c8-93dd-aba2e5b2ec38)

**Comment by xulihang** [comment]
The plugin does not rely on dwt.azurewebsites.net. It is for uploading files and you can host your own server.

To use the plugin in an offline environment, you need to host a copy of the js library in `https://cdn.jsdelivr.net/npm/dwt@latest/dist/` by yourself and modify path in the javascript.
